.cirun.yml

```
@@ -0,0 +1,9 @@
+runners:
+  - name: aws-gpu-runner
+    cloud: aws
+    instance_type: g4dn.xlarge
+    machine_image: ami-0678adbdcb4c3a662
+    preemptible: false
+    workflow: .github/workflows/build_gpu.yml
+    labels:
+      - aws-gpu
```

```suggestion
runners:
  - name: aws-gpu-runner
    cloud: aws
    # Cheapest machine we can attach a GPU to
    instance_type: g4dn.xlarge
    # Machine image, as used in https://docs.cirun.io/custom-images/cloud-custom-images#using-prebuilt-images-nvidia-deep-learning-ami
    # https://aws.amazon.com/marketplace/pp/prodview-7ikjtg3um26wq?sr=0-1&ref_=beagle&applicationId=AWS-EC2-Console
    machine_image: ami-0678adbdcb4c3a662
    preemptible: false
    workflow: .github/workflows/build_gpu.yml
    labels:
      - cirun-aws-gpu
```

The label may need to start with `cirun`:  https://docs.cirun.io/reference/yaml#labels-labels ?

---

But it seems like it's working without?

---

It works for now, because you have `self-hosted` in your workflow yaml, but its deprecated. I would recommend using label name starting with cirun to avoid breaking it in future.

I would also recommend suffixing the `runs-on` param with `--${{ github.run_id }}"` to ensure that the runners are consumed by the workflow which created it.

See: https://docs.cirun.io/reference/unique-runner-labels

TLDR:
```yml
runs-on: "cirun-aws-gpu--${{ github.run_id }}"
```

In your `.cirun.yml`:
```yml
    labels:
      - cirun-aws-gpu
```

---

Thank you!